src/VisualStudio/Core/Def/Implementation/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -355,30 +356,29 @@ public async Task UpdateBufferContentsAsync(CancellationToken cancellationToken)
                 }
                 else
                 {
+                    var generatedDocuments = await project.GetSourceGeneratedDocumentsAsync(cancellationToken).ConfigureAwait(false);
+                    var generatedDocument = generatedDocuments.SingleOrDefault(d =>
+                                d.SourceGenerator.GetType().FullName.Equals(_generatorTypeName, StringComparison.OrdinalIgnoreCase) &&
+                                d.SourceGenerator.GetType().Assembly.GetName().Name.Equals(_generatorAssemblyName));
 
-                    var generatorDriverRunResults = await project.GetGeneratorDriverRunResultAsync(cancellationToken).ConfigureAwait(false);
-                    var generatorRunResult = generatorDriverRunResults?.Results.SingleOrNull(r =>
-                                r.Generator.GetType().FullName.Equals(_generatorTypeName, StringComparison.OrdinalIgnoreCase) &&
-                                r.Generator.GetType().Assembly.GetName().Name.Equals(_generatorAssemblyName));
-
-                    if (generatorRunResult == null)
+                    if (generatedDocument != null)
                     {
-                        windowFrameMessageToShow = string.Format(ServicesVSResources.The_generator_0_that_generated_this_file_has_been_removed_from_the_project, _generatorTypeName);
-                        windowFrameImageMonikerToShow = KnownMonikers.StatusError;
+                        windowFrameMessageToShow = string.Format(ServicesVSResources.This_file_is_autogenerated_by_0_and_cannot_be_edited, _generatorTypeName);
```

Comment:
can you show a picture of this?

Reply:
![image](https://user-images.githubusercontent.com/201340/99749480-7ebff300-2a93-11eb-9fcf-00f984ed17fe.png)